Integrations/URLHaus/URLHaus.py

```diff
@@ -0,0 +1,493 @@
+            demisto.results({
+                'Type': entryTypes['note'],
+                'ContentsFormat': formats['json'],
+                'Contents': file_information,
+                'HumanReadable': human_readable,
+                'HumanReadableFormat': formats['markdown'],
+                'EntryContext': ec
+            })
+        elif (file_information['query_status'] == 'ok' and not file_information['md5_hash']) or \
+                file_information['query_status'] == 'no_results':
+            human_readable = f'## URLhaus reputation for {hash_type.upper()} : {hash}\n' \
+                f'No results!'
+
+            demisto.results({
+                'Type': entryTypes['note'],
+                'ContentsFormat': formats['json'],
+                'Contents': file_information,
+                'HumanReadable': human_readable,
+                'HumanReadableFormat': formats['markdown'],
+            })
+        elif file_information['query_status'] in ['invalid_md5', 'invalid_sha256']:
+            human_readable = f'## URLhaus reputation for {hash_type.upper()} : {hash}\n' \
+                f'Invalid {file_information["query_status"].lstrip("invalid_").upper()}!'
+
+            demisto.results({
+                'Type': entryTypes['note'],
+                'ContentsFormat': formats['json'],
+                'Contents': file_information,
+                'HumanReadable': human_readable,
+                'HumanReadableFormat': formats['markdown'],
+            })
+        else:
+            demisto.results({
+                'Type': entryTypes['error'],
+                'ContentsFormat': formats['text'],
+                'Contents': f'Query results = {file_information["query_status"]}'
+            })
+
+    except Exception:
+        print(traceback.format_exc())
+        demisto.debug(traceback.format_exc())
+        return_error('Failed getting file data, please verify the arguments and parameters')
+
+
+def urlhaus_download_sample_command():
+    file_sha256 = demisto.args()['file']
+    res = download_malware_sample(file_sha256)
+
+    try:
+        if len(res.content) == 0:
+            demisto.results({
+                'Type': entryTypes['note'],
+                'HumanReadable': f'No results for SHA256: {file_sha256}',
+                'HumanReadableFormat': formats['markdown']
+            })
+        elif res.json()['query_status'] == 'not_found':
+            demisto.results({
+                'Type': entryTypes['note'],
+                'ContentsFormat': formats['json'],
+                'Contents': res.json(),
+                'HumanReadable': f'No results for SHA256: {file_sha256}',
+                'HumanReadableFormat': formats['markdown']
+            })
+        else:
+            demisto.results({
+                'Type': entryTypes['error'],
+                'ContentsFormat': formats['text'],
+                'Contents': res.content
+            })
+    except json.JSONDecodeError:
```

Comment:
This is how we save the sample? Why do we handle this flow with except?

Reply:
Changed